src/storages/storage_manager.js

```diff
@@ -25,42 +29,39 @@ const MAX_OPENED_STORAGES = 1000;
  * @property {LruCache} cache
  * @ignore
  */
-class StorageManager {
+export default class StorageManager {
     /**
      * @param {T} StorageConstructor
+     * @param {Configuration} [config]
      */
-    constructor(StorageConstructor) {
+    constructor(StorageConstructor, config = Configuration.getGlobalConfig()) {
         this.StorageConstructor = StorageConstructor;
         this.name = StorageConstructor.name;
-        this.cache = cacheContainer.openCache(this.name, MAX_OPENED_STORAGES);
+        this.cache = cacheContainer.openCache(this.name, config.get('maxOpenedStorages'));
```

Comment:
I still wonder whether the `maxOpenedStorages` needs to be configurable or not. It's really just a failsafe. I don't even know what would happen if the cache would overflow.

Reply:
Well you'd probably reach the memory limits and the node process would fail - if you have reference to the values, they cannot be garbage collected. But it would have to be really large number of instances.

I will leave "whether it is needed or not" to you, as I am not sure what are the real numbers we can see in production. I just saw another constant so made it configurable, we can keep it inlined if it does not make sense to configure it, but it also does not hurt having all the options in a single place.

Reply:
I meant if the storages cache would overflow and start deleting old storages. I don't think they're entirely stateless, so losing the instance and recreating it again might have an impact.

Let's not make it configurable. The typical number of storages is 3, in extreme cases there might be tens of them. I don't think I've ever seen a hundred.

Reply:
In that case I'd even suggest removing the caching, as it feels unnecessary. Even tens of thousands would be most probably ok. 

Apparently the `cacheContainer` is used only here, so we could remove that as well. Less code to maintain, less code to break...

Reply:
The caching is important so that you don't crate new datasets on Apify (e.g. making more API calls than needed).

Reply:
I am talking about the `cacheContainer` with LRU cache, simple `Map` would do the same job, right?